homeassistant/components/number/__init__.py

```diff
@@ -55,8 +55,249 @@
 class NumberDeviceClass(StrEnum):
     """Device class for numbers."""
 
-    # temperature (C/F)
+    # NumberDeviceClass should be aligned with SensorDeviceClass
```

Comment:
I'm quite worried that this won't stay align.
Is there anything we can do to make it automatic?
Could SensorDeviceClass inherit NumberDeviceClass maybe?

Reply:
Can we break out the number device class items from the SensorDeviceClass into a separate parent class that both SensorDeviceClass and NumberDeviceClass can inherit?

Reply:
After investigation, inheritance is not recommended for Enum:
https://docs.python.org/3.10/library/enum.html#restricted-enum-subclassing

So I think tests is the best way to address this.

Reply:
I added a couple of tests, they only check the name, not the value though. Do we need to check both?

Reply:
@akx the two Enum are not 100% aligned (Sensor has a few extras which need to be taken into consideration)
@emontnemery I think we need to check the values also, as that will be what's used with automatic conversion etc.

Reply:
There is no disaster going on if this would be missed at some point. The comment is sufficient at this point.

With the additional tests added in this PR I think it fine as is now.

Reply:
The tests are updated so they check both name and value of the enums